### PR TITLE
[Pallas] Introduce gmm_backward

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request introduces a helper for gmm_backward. I'm still debuting if we need to make gmm as a autograd.function given we will do manual back-propagation in Mixtral.

Test Plan:
python test/test_gmm.py

### CODE REVIEW

File: test/test_gmm.py [hunk 1]
```diff
@@ -344,6 +344,36 @@ def test_tgmm_bf16(self):
     # Make sure tgmm doesn't fallback.
     self.assertNotIn("aten::", met.short_metrics_report())
 
+  @unittest.skipIf(xr.device_type() != 'TPU', "This test only works on TPU.")
+  def test_gmm_backward(self):
+    self._init_test_cases()
```
Comment (JackCaoG) [inline test/test_gmm.py:349]: nit, you might need a `met.clear_all()` here.

**Review by JackCaoG — APPROVED** [review]
What do you mean by "manual backprop in mixtral"?

File: test/test_gmm.py (hunk 1 above)
Comment (alanwaketan) [reply test/test_gmm.py:349]: Let me fix it in the next PR. Don't want to waste CI cycles.

**Comment by alanwaketan** [comment]
Skip GPU tests to move fast.

**Comment by alanwaketan** [comment]
> What do you mean by "manual backprop in mixtral"?

Just we need to override the MoE back prop to accommondate gmm backward and manual sharding. You will know what that means once the code is ready. Thanks for approving this change.

**Pull request merged**